ros_ws/src/car_control/src/dms_controller.cpp

```diff
@@ -63,6 +60,20 @@ void DMSController::publishDriveMode()
     this->m_drive_mode_publisher.publish(drive_mode_message);
 }
 
+bool DMSController::getEmergencyStop()
+{
+    auto current_time = ros::Time::now();
+    return (!this->m_last_emergencystop.is_zero() &&
+            this->m_last_emergencystop + this->m_emergencystop_expiration_time > current_time);
```

Comment:
Please also check if `this->m_last_emergencystop < current_time`. Otherwise, this creates false positives if the simulation is reset.